src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -844,8 +844,23 @@ private SNINativeMethodWrapper.ConsumerInfo CreateConsumerInfo(bool async)
             return myInfo;
         }
 
-        internal void CreatePhysicalSNIHandle(string serverName, bool ignoreSniOpenTimeout, long timerExpire, out byte[] instanceName, byte[] spnBuffer, bool flushCache,
-                bool async, bool fParallel, TransparentNetworkResolutionState transparentNetworkResolutionState, int totalTimeout, SqlConnectionIPAddressPreference ipPreference, string cachedFQDN, bool isTDS8, string hostNameInCertificate)
+        internal void CreatePhysicalSNIHandle(
+            string serverName,
+            bool ignoreSniOpenTimeout,
+            long timerExpire,
+            out byte[] instanceName,
+            byte[] spnBuffer,
+            bool flushCache,
+            bool async,
+            bool fParallel,
+            TransparentNetworkResolutionState transparentNetworkResolutionState,
+            int totalTimeout,
+            SqlConnectionIPAddressPreference ipPreference,
+            string cachedFQDN,
+            bool tlsFirst = false,
+            string hostNameInCertificate = "",
+            string databaseName = "",
+            ApplicationIntent applicationIntent = ApplicationIntent.ReadWrite)
```

Comment:
This is a lot of parameters. It might be worth just making a class to hold them.